### PR TITLE
Major release 6.0.0

### PR DESCRIPTION
**Pull request by laura-rodriguez**
- Add support for OASv3
- Regenerate code using OpenAPI generator tech
- Update readme, changelog and migrating guide (REVIEWERS pls take another look at these docs)
- All the changes have been already approved before

### CODE REVIEW

**Review by arvindkrishnakumar-okta — APPROVED** [review]
lgtm 🚢

File: README.md [hunk 1]
```diff
@@ -87,51 +83,79 @@ Visual Studio 2017 or newer is required as previous versions are not compatible
...
 * An [API token](https://developer.okta.com/docs/api/getting_started/getting_a_token)
  
-### Initialize a client 
-Construct a client instance by passing it your Okta domain name and API token:
- 
-``` csharp
-var client = new OktaClient(new OktaClientConfiguration
+### Initialize an API client 
+
+Construct an API client instance by it your Okta domain and API token.
```
Comment (bryanapellanes-okta) [inline README.md:93]: I think this sentence looks incomplete.  Not sure what is being said here.

**Review by bryanapellanes-okta — COMMENTED** [review]
One observation of something that should probably be fixed.

**Review by bryanapellanes-okta — APPROVED** [review]
🚢

**Pull request merged**